### PR TITLE
Ports Medbot Improvement + Refactor

### PR DESCRIPTION
**Pull request by Birdtalon**
I wrote this for Paradise, however I received no feedback there. Feel free to use if you wish.

Refactors parts of the medbot code, breaks a couple of huge procs into smaller ones, as well as adding some intelligence to the medbots themselves.

- Medbots now check which type of damage is greatest for the patient and will treat that first.
- When using an internal beaker, medbots will check to see if the reagent is actually useful for treating the damage before using it. If not it will fall back on the internal synthesizer .
- Medbots will no longer inject their internal beaker with impunity regardless of if it's helpful or harmful. They must be toggled to over-ride this manually.
- Medbots will now only seek to treat viruses if the patient is reasonably healthy first (over 50 health) and even then will prioritise any serious damage first before attempting to treat the virus.

### CODE REVIEW

File: code/_globalvars/lists/reagents.dm [hunk 1]
```diff
@@ -66,3 +66,10 @@ var/global/list/safe_chem_list = list("antihol", "charcoal", "epinephrine", "ins
 									  "omnizine", "stimulants", "synaptizine", "potass_iodide", "oculine", "mannitol", "styptic_powder",
 									  "spaceacillin", "salglu_solution", "sal_acid", "cryoxadone", "blood", "synthflesh", "hydrocodone",
 									  "mitocholide", "rezadone")
+
+//Lists of chems by what they heal for INJEST only
+
+var/global/list/brute_healing_chems_injest = list("salglu_solution","atropine","sal_acid","omnizine","epinephrine")
+var/global/list/burn_healing_chems_injest = list("salglu_solution","atropine","omnizine","epinephrine")
+var/global/list/oxy_healing_chems_injest = list("salbutamol","atropine","perfluorodecalin","epinephrine")
+var/global/list/tox_healing_chems_injest = list("charcoal","omnizine","pen_acid","epinephrine")
```
Comment (msgerbs) [inline code/_globalvars/lists/reagents.dm]: "Injest"

File: code/modules/mob/living/simple_animal/bot/construction.dm [hunk 2]
```diff
@@ -292,6 +292,7 @@
 	var/treatment_fire = "salglu_solution"
 	var/treatment_tox = "charcoal"
 	var/treatment_virus = "spaceacillin"
+	var/treatment_evil = "pancuronium"
```
Comment (msgerbs) [inline code/modules/mob/living/simple_animal/bot/construction.dm:295]: Would it be possible/beneficial to refactor this in a similar way as the build step?

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 3]
```diff
@@ -547,11 +548,11 @@
 	switch(skin)
 		if("ointment")
 			new /obj/item/weapon/storage/firstaid/fire/empty(Tsec)
-		if("tox")
+		if(TOX)
 			new /obj/item/weapon/storage/firstaid/toxin/empty(Tsec)
 		if("o2")
 			new /obj/item/weapon/storage/firstaid/o2/empty(Tsec)
-		if("brute")
+		if(BRUTE)
```
Comment (msgerbs) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: I don't think you meant to change these.

**Review by msgerbs — CHANGES_REQUESTED** [review]
See comments

File: code/modules/mob/living/simple_animal/bot/medbot.dm (hunk 3 above)
Comment (Birdtalon) [reply code/modules/mob/living/simple_animal/bot/medbot.dm]: The define `BRUTE` is `"brute"` so it makes sense to use the existing defines.

File: code/_globalvars/lists/reagents.dm (hunk 1 above)
Comment (Birdtalon) [reply code/_globalvars/lists/reagents.dm]: You want it with a capital I?

File: code/modules/mob/living/simple_animal/bot/construction.dm (hunk 2 above)
Comment (Birdtalon) [reply code/modules/mob/living/simple_animal/bot/construction.dm:295]: How do you mean?

**Comment by msgerbs** [comment]
It doesn't compile :(

**Review by msgerbs — APPROVED** [review]

**Pull request merged**